.github/workflows/examples-components.yml

```diff
@@ -117,3 +132,86 @@ jobs:
       - name: build project
         run: wash build
         working-directory: examples/${{ matrix.project.lang }}/components/${{ matrix.project.name }}
+      # Save example as an artifact for later step(s)
+      - uses: actions/upload-artifact@v4
+        if: ${{ startswith(github.ref, format('refs/tags/component-{0}-v', matrix.project.name)) }}
+        with:
+          name: wash-build-${{ matrix.wash-version }}-${{ matrix.project.lang }}-component-${{ matrix.project.name }}
+          path: examples/${{ matrix.project.lang }}/components/${{ matrix.project.name }}/build/${{ matrix.project.wasm-bin }}
+
+  # Publish components relevant components if they've been tagged
+  publish:
+    name: publish
+    runs-on: ubuntu-22.04
+    needs: [ wash-build ]
+    if: ${{ startswith(github.ref, 'refs/tags/component-') }}
+    strategy:
+      fail-fast: false
+      matrix:
+        wash-version:
+          - current
+        project:
+          - lang: "rust"
+            name: "http-jsonify"
+            wasm-bin: "wasmcloud_component_http_jsonify_s.wasm"
+    steps:
+      # Determine tag version (if this is a release tag), without the 'v'
+      - name: Determine version
+        id: meta
+        shell: bash
+        env:
+          TAG: ${{ github.ref_name }}
+          BRANCH_PREFIX: ${{ format('refs/tags/component-{0}-v', matrix.project.name) }}
```

Comment:
I'm having a hard time narrowing it down, but looking at the matrix here am I right in thinking that this will release the http-jsonify component every time this action is triggered when the tag starts with `component-`?

Reply:
Yup, and for every step after that which matters it should narrow it down to the actual branch for the single component that matches in the matrix -- see the `if`s that skip afterwards, depending on `branch-prefix`

Reply:
Ah so that's the part. Okay cool 👍🏻 Branch prefix is what I needed

Reply:
Yeah I will agree it's a bit hard to understand...

One thing I could do is change this to break apart/parse the tag and do it that way? We just need to be very specific about tag names and then it would be good?